### PR TITLE
Add Cimpress.Nancy.Demo

### PR DESCRIPTION
**Pull request by yahehe**
Sample project for #10

### CODE REVIEW

File: demo/Cimpress.Nancy.Demo/Properties/launchSettings.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      "launchBrowser": true,
+      "launchUrl": "api/values",
+      "environmentVariables": {
+        "ASPNETCORE_ENVIRONMENT": "Development"
+      }
+    },
+    "Cimpress.Nancy.Demo": {
+      "commandName": "Project",
+      "launchBrowser": true,
+      "launchUrl": "http://localhost:5000/api/values",
```
Comment (jnallard) [inline demo/Cimpress.Nancy.Demo/Properties/launchSettings.json]: Was this the default?

File: demo/Cimpress.Nancy.Demo/DemoNancyServiceBootstrapper.cs [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    public class DemoNancyServiceBootstrapper : NancyServiceBootstrapper
+    {
+        public DemoNancyServiceBootstrapper() : base()
+        {
+            
+        }
+
+        protected override void ApplicationStartup(TinyIoCContainer container, IPipelines pipelines)
+        {
+            container.Register<INancyLogger, DemoLogger>();
```
Comment (jnallard) [inline demo/Cimpress.Nancy.Demo/DemoNancyServiceBootstrapper.cs]: Do you need this? 
If I recall, the register with default function will register whatever class implements INancyLogger for you.

File: demo/Cimpress.Nancy.Demo/DemoLogger.cs [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    public class DemoLogger : INancyLogger
+    {
+        public void Configure(IDictionary<string, string> options)
+        {
+
+        }
+
+        public void Debug(object data)
+        {
+            Console.WriteLine($"Debug: {data}");
```
Comment (jnallard) [inline demo/Cimpress.Nancy.Demo/DemoLogger.cs]: When I ran the demo, I couldn't see anything being printed to the output window in VS. Where should it go?

**Review by jnallard — COMMENTED** [review]
Would it make sense to call demo 'demos' or 'samples', so we can easily add more later. 
I think it would be nice to show how the different submodules can work independently. (A separate PR though)

Can you add a Module or something, so there's an actual endpoint to test? You created a VersionModuleExtender, but no VersionModule to extend.

File: demo/Cimpress.Nancy.Demo/Properties/launchSettings.json (hunk 1 above)
Comment (yahehe) [reply demo/Cimpress.Nancy.Demo/Properties/launchSettings.json]: Yeah, I can put this in the .gitignore

File: demo/Cimpress.Nancy.Demo/DemoNancyServiceBootstrapper.cs (hunk 2 above)
Comment (yahehe) [reply demo/Cimpress.Nancy.Demo/DemoNancyServiceBootstrapper.cs]: It does, I chose to do it explicitly to make it clearer what's happening

File: demo/Cimpress.Nancy.Demo/DemoLogger.cs (hunk 3 above)
Comment (yahehe) [reply demo/Cimpress.Nancy.Demo/DemoLogger.cs]: IIRC VS is weird with console output, not sure

File: demo/Cimpress.Nancy.Demo/Properties/launchSettings.json (hunk 1 above)
Comment (jnallard) [reply demo/Cimpress.Nancy.Demo/Properties/launchSettings.json]: No, I just meant the launchUrl setting!
You should have it just be the '/' endpoint, or something from a module.
I think you need this file for Asp.Net?

**Comment by jnallard** [comment]
I meant that we should rename the folder from demo to demos or samples, not the project itself, so that we could have a directory structure like:

-samples
--Cimpress.Nancy.Demo
--Cimpress.Nancy.Security.Demo
-- . . . etc
-src
--Cimpress.Nancy
-- . . . etc
-test
--Cimpres.Nancy.Tests
-- . . . etc

**Comment by yahehe** [comment]
That is how it's set up?

**Comment by jnallard** [comment]
It looks like you renamed the project to "demos" and not the parent folder?

demo

**Comment by jnallard** [comment]
I miss Stash. I want to do file comments!

Anyways, why did you remove the auto start of loading a webpage? I think it was pretty helpful for someone running the demo. 

Also, I tried to connect to "http://localhost:43304/v0/demo", but I only got 404s?

**Comment by yahehe** [comment]
The webpage is less applicable when you're writing an API. Also pretty annoying when every time you start it pulls up your browser. The launchsettings can always be edited locally

**Comment by yahehe** [comment]
Should work now

**Review by jnallard — APPROVED** [review]
Tested and it worked (At least got the authentication error)

**Pull request merged**